### PR TITLE
Hub browser context menu

### LINKED ISSUE #14: Refactor/Move the deploy script
Refactor or move to the MD adapter repository the deploy script that targets the MagicDraw adapter

### PR DESCRIPTION
**Pull request by nathanatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/DEH-CommonJ/pulls) open
- [x] I have verified that I am following the DEH-CommonJ [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/DEH-CommonJ/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

closes #14 and #16 
- Small changes while building the MD adapter and the Capella adapter
- Added the annotation exclude from code coverage result
- Added a context menu for the Hub browsers and the interface for the view model that is to be implemented in the adapters
- Updated the deploy script so it fits for any adapter that may use it

### CODE REVIEW

File: src/main/java/Services/MappingEngineService/MappingEngineService.java [hunk 1]
```diff
@@ -183,10 +183,9 @@ private void PopulateRules(String ruleAssembly)
...
         
-        Set<Class<? extends IMappingRule>> mappingRules = reflections.getSubTypesOf(MappingRule.class)
+        return reflections.getSubTypesOf(MappingRule.class)
                 .stream()
+                .filter(x -> !java.lang.reflect.Modifier.isAbstract(x.getModifiers()))
                 .collect(Collectors.toSet());
-        
-        return mappingRules;
     }
 }
```
Comment (antoineatrhea) [inline src/main/java/Services/MappingEngineService/MappingEngineService.java:191]: Missing EOF

File: src/main/java/Services/MappingEngineService/MappingRule.java [hunk 2]
```diff
@@ -40,17 +45,10 @@
...
-     * Transforms a object of type {@linkplain TInput} to another one of type {@linkplain TOutput}
-     * 
-     * @param input the input object to transform
-     * @return a {@linkplain TOutput} object
-     */
-    @Override
-    public abstract TOutput Transform(Object input);
+    protected Logger Logger = LogManager.getLogger();
     
+    private IMappingConfigurationService mappingConfiguration;
```
Comment (antoineatrhea) [inline src/main/java/Services/MappingEngineService/MappingRule.java:50]: Missing doc

File: src/main/java/ViewModels/ObjectBrowser/RequirementTree/Rows/IterationRequirementRowViewModel.java [hunk 3]
```diff
@@ -64,6 +64,7 @@ public void ComputeContainedRows()
         
         this.GetThing().getRequirementsSpecification()
             .stream()
+            .filter(x -> !x.isDeprecated())
             .forEach(x -> this.containedRows.add(new RequirementSpecificationRowViewModel(x, this)));
     }    
 }
```
Comment (antoineatrhea) [inline src/main/java/ViewModels/ObjectBrowser/RequirementTree/Rows/IterationRequirementRowViewModel.java:70]: Missing EOF

File: src/main/java/ViewModels/Rows/MappedElementRowViewModel.java [hunk 4]
```diff
@@ -226,4 +282,58 @@ public boolean GetShouldCreateNewTargetElementValue()
...
+        if(this.GetMappingDirection() == mappingDirection)
+        {
+           return this.GetElementRepresentation(representation);
+        }        
+
+        return representation;
+    }
+    
+    /**
+     *
```
Comment (antoineatrhea) [inline src/main/java/ViewModels/Rows/MappedElementRowViewModel.java:327]: Empty doc

File: src/main/java/ViewModels/Rows/MappedElementRowViewModel.java [hunk 5]
```diff
@@ -226,4 +282,58 @@ public boolean GetShouldCreateNewTargetElementValue()
...
+     * @return a {@linkplain String}
+     */
+    public String GetHubElementRepresentation(Class<? extends Thing> thingType)
+    {
+        return this.GetElementRepresentation(this.GetHubElement() == null ? "-" : this.GetHubElement().getName(),
+                thingType.getSimpleName(), MappingDirection.FromDstToHub);
+    }
+    
+    /**
+     * @param element
```
Comment (antoineatrhea) [inline src/main/java/ViewModels/Rows/MappedElementRowViewModel.java:300]: Empty doc

File: src/main/java/Views/package-info.java [hunk 6]
```diff
@@ -21,7 +21,4 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-/**
- *
- */
 package Views;
```
Comment (antoineatrhea) [inline src/main/java/Views/package-info.java:24]: Missing EOF

**Review by antoineatrhea — CHANGES_REQUESTED** [review]

**Review by alexatrhea — APPROVED** [review]
approved pending antoine's comments

**Review by antoineatrhea — APPROVED** [review]

**Pull request merged**